lib/ts/recipe/multifactorauth/recipe.ts

```diff
@@ -258,4 +260,52 @@ export default class Recipe extends RecipeModule {
         }
         return result;
     };
+
+    async checkIfLinkingAllowed(session: SessionContainerInterface, factorIds: string[], userContext: UserContext) {
+        logDebugMessage("checkIfLinkingAllowed called");
+        let caughtSetupFactorError;
+        const mfaInfo = await getMFARelatedInfoFromSession({
+            session,
+            userContext,
+        });
+        const validFactorIds = [];
+        // In all apis besides passwordless createCode, we know exactly which factor we are signing into, so in those cases,
+        // this is basically just checking if the single factor is allowed to be setup or not.
+        // For createCode (if the FE didn't pass the factor id exactly, which it should for MFA),
+        // we filter whatever is allowed. If any of them are allowed, createCode can happen.
+        // The filtered list can be used to select email templates. As an example:
+        // If the flowType for passwordless is USER_INPUT_CODE_AND_MAGIC_LINK and but only the otp-email factor is allowed to be set up
+        // then we do not want to include a link in the email.
+        for (const id of factorIds) {
+            logDebugMessage(`checkIfLinkingAllowed checking assertAllowedToSetupFactorElseThrowInvalidClaimError`);
+            try {
+                await this.recipeInterfaceImpl.assertAllowedToSetupFactorElseThrowInvalidClaimError({
```

Comment:
this function is only called if we are linking something to the session user (as shown by the function name).
We do not check for signup, this is is called whenever a new link will be created.
I think doing this based on the linking status is better since,
- this covers both sign in and up naturally
- it runs this check every time the session user would be changed.
From the MFA perspective linking is the same as factor setup. I've updated the comment where this function is called to say that the link creation is essentially the same as MFA setup.